File: src/POS.Application/Commons/Base/BaseResponse.cs
namespace POS.Application.Commons.Base;

public class BaseResponse<T>
{
    public bool Success { get; set; }
    public T? Data { get; set; }
    public string? Message { get; set; }
    public BaseResponseError? Errors { get; set; }

    public BaseResponse()
    {
        Errors = new BaseResponseError();
    }
}

File: src/POS.Api/ClientApp/src/Commons/Base/BaseResponse.ts
export interface BaseReponse<T, U> {
    success?: boolean
    message?: string
    data: T
    errors: U
}

export interface BaseRegister {
    success: boolean
    data: boolean
    message: string
    errors: Errors
}

export interface Errors {
    login: any
    register: Register
}

export interface Register {
    email: string[]
    password: string[]
    firstName: string[]
    lastName: string[]
}

export interface IRegisterResponseDto {
    register: {
        email: Array<string>
        password: Array<string>
        firstName: Array<string>
        lastName: Array<string>
    }
}
